op-batcher/batcher/channel_builder_test.go

```diff
@@ -618,3 +632,75 @@ func TestFramePublished(t *testing.T) {
 	// Now the timeout will be 1000
 	require.Equal(t, uint64(1000), cb.timeout)
 }
+
+func TestChannelBuilder_InputBytes(t *testing.T) {
+	require := require.New(t)
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	cb, _ := defaultChannelBuilderSetup(t)
+
+	require.Zero(cb.InputBytes())
+
+	var l int
+	for i := 0; i < 5; i++ {
+		block := newMiniL2Block(rng.Intn(32))
+		l += blockBatchRlpSize(t, block)
+
+		_, err := cb.AddBlock(block)
+		require.NoError(err)
+		require.Equal(cb.InputBytes(), l)
+	}
+}
+
+func TestChannelBuilder_OutputBytes(t *testing.T) {
+	require := require.New(t)
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	cfg := defaultTestChannelConfig
+	cfg.TargetFrameSize = 1000
+	cfg.MaxFrameSize = 1000
+	cfg.TargetNumFrames = 16
+	cfg.ApproxComprRatio = 1.0
+	cb, err := newChannelBuilder(cfg)
+	require.NoError(err, "newChannelBuilder")
+
+	require.Zero(cb.OutputBytes())
+
+	for {
+		block, _ := dtest.RandomL2Block(rng, rng.Intn(32))
+		_, err := cb.AddBlock(block)
+		if errors.Is(err, ErrInputTargetReached) {
+			break
+		}
+		require.NoError(err)
+	}
+
+	require.NoError(cb.OutputFrames())
+	require.True(cb.IsFull())
+	require.Greater(cb.NumFrames(), 1)
+
+	frames := make([]frameData, 0, cb.NumFrames())
+	var flen int
+	for cb.HasFrame() {
+		f := cb.NextFrame()
+		frames = append(frames, f)
+		flen += len(f.data)
+	}
+
+	require.Equal(cb.OutputBytes(), flen)
+}
+
+func defaultChannelBuilderSetup(t *testing.T) (*channelBuilder, ChannelConfig) {
+	t.Helper()
```

Comment:
Personally think it doesn't make sense to pass testing into helpers at all. Just bubble up the error and allow each test to handle respectively, but this is ok

Reply:
It's a common pattern, which is why there's `t.Helper()` so that helper functions are not part of a stack trace when an error happens. If there's a common way of handling an error, you save some repetition in each calling test by just doing the right assertions in the helper function.